src/wasm-traversal.h

```diff
@@ -247,7 +247,8 @@ struct Walker : public VisitorType {
       self->visitExport(curr.get());
     }
     for (auto& curr : module->globals) {
-      self->walkGlobal(curr.get());
+      // TODO: Stop parsing from inserting nullptr into globals
```

Comment:
this looks very bad. what's going on here?

Reply:
We need to figure that out. Instead of adding this hack in the code, how about disabling the wasm2asm test on it? Hacking a test seems better, even if this is temporary. And lets file an issue for that problem.

Reply:
It turns out that the root issue was actually that the vector of globals was modified while being iterated through, which invalidated the iterator.

Reply:
I see, nice fix.